docs/user-guide/gateway.md

```diff
@@ -0,0 +1,125 @@
+# Gateway
+
+Not all network services can be provided by the fabric itself due to limitations with modern switch hardware and software.
+The Gateway is designed to provide these more sophisticated network services, such as NAT, PAT, firewalling, and others.
+The tradeoff is that while simple [VPC Peerings](vpcs.md#vpcpeering) offer the full cut-through bandwidth of the fabric, gateway services are limited by the traffic handling capability of the gateway nodes.
+
+
+## Gateway Nodes and Fabric Connectivity
+
+Gateway nodes are connected to the fabric by a set of physical connectionsthat are modeled via Connection objects.
+See the section on [Gateway Connections](connections.md#gateway-connections) for connection requirements and example configuration.
+
+When a Gateway Peering is used to connect two VPCs or externals, the appropriate gateway nodes will attract traffic to themselves by advertising the appropriate routes to the fabric.
+In turn, the fabric will use these routes to steer traffic to the gateway so that it can apply the configured peering policy.
+
+Gateway nodes use BGP to advertise routes to the fabric, and the gateway will get its own ASN so it is possible to easily identify gateway advertised routes in the fabric.
+
+## Gateway Peering
+
+Just as [VPC Peerings](vpcs.md#vpcpeering) provide VPC to VPC connectivity by way of the switches in the fabric, gateway peerings provide connectivity via the gateway nodes.
+Gateway services can be inserted between any pair of VPCs or externals using a Gateway Peering.
```

Comment:
peering between VPCs and externals depend on merging the externals-as-VPC work that has not been reviewed yet, so it is not clear to me that we're targeting it for the next release. Peering an external to another external is something we have briefly discussed but not really tested or attempted to tackle, and which is even less likely to make it for the next release